client/src/components/chat.tsx

```diff
@@ -94,163 +91,220 @@ function ChatThread(props: {
+                  <ThumbUpIcon />
+                </Avatar>
+              </ListItemAvatar>
+            </div>
+            <div
+              id="click-neutral"
+              onClick={() => {
+                if (message.feedbackId) {
+                  handleSelectFeedback(message.feedbackId, Feedback.NEUTRAL);
+                }
               }}
-              transformOrigin={{
-                vertical: "center",
-                horizontal: "left",
+            >
+              <ListItemAvatar>
+                <Avatar className={styles.avatar}>
+                  <ThumbsUpDownIcon />
+                </Avatar>
+              </ListItemAvatar>
+            </div>
+            <div
+              id="click-bad"
+              onClick={() => {
+                if (message.feedbackId) {
+                  handleSelectFeedback(message.feedbackId, Feedback.BAD);
+                }
               }}
-              elevation={0}
-              className={styles.popover}
             >
-              <div
-                onClick={() => {
-                  if (message.feedbackId) {
-                    handleSelectFeedback(i, message.feedbackId, Feedback.GOOD);
-                  }
-                }}
-              >
-                <ListItemAvatar>
-                  <Avatar
-                    className={styles.avatar}
-                    style={{ background: "green" }}
-                  >
-                    <ThumbUpIcon />
-                  </Avatar>
-                </ListItemAvatar>
-              </div>
-              <div
-                onClick={() => {
-                  if (message.feedbackId) {
-                    handleSelectFeedback(i, message.feedbackId, Feedback.BAD);
-                  }
-                }}
-              >
-                <ListItemAvatar>
-                  <Avatar
-                    className={styles.avatar}
-                    style={{ background: "red" }}
-                  >
-                    <ThumbDownIcon />
-                  </Avatar>
-                </ListItemAvatar>
-              </div>
-            </Popover>
-          </ListItem>
-        );
-      })}
-    </List>
+              <ListItemAvatar>
+                <Avatar className={[styles.avatar, styles.BAD].join(" ")}>
+                  <ThumbDownIcon />
+                </Avatar>
+              </ListItemAvatar>
+            </div>
+          </div>
+        </div>
+      </Popover>
+    </ListItem>
   );
 }
 
-function Chat(props: { height: number; search: any }): JSX.Element {
+function Chat(props: { height: number }): JSX.Element {
   const styles = useStyles();
-  const state = useSelector<State, State>(state => state);
-  const [messages, setMessages] = useState<ChatMsg[]>([]);
-  const [lastQuestionAt, setLastQuestionAt] = useState<Date>();
-  const [lastAnswerAt, setLastAnswerAt] = useState<Date>();
+  const [chatData, setChatData] = useState<ChatData>({
```

Comment:
this is fine as useState and not useSelector?

Reply:
I would like to clean this up in a future pass and move the state to redux, but for now it's passing tests as is